elide-datastore/elide-datastore-hibernate/src/main/java/com/yahoo/elide/core/filter/CaseAwareJPQLGenerator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.core.filter;
+
+import com.yahoo.elide.core.exceptions.InvalidValueException;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * JPQLPredicateGenerator that can generate both case sensitive/insensitive predicate fragments.
+ */
+public class CaseAwareJPQLGenerator implements FilterTranslator.JPQLPredicateGenerator {
+
+    private static final String COMMA = ", ";
+    private static final String FILTER_PATH_NOT_NULL = "Filtering field path cannot be empty.";
+    private static final String FILTER_ALIAS_NOT_NULL = "Filtering alias cannot be empty.";
+
+    private Case upperOrLower;
+    private String jpqlTemplate;
+    private ArgumentCount argumentCount;
+
+    /**
+     * Whether to use uppercase, lowercase, or not transformation for case insensitive queries.
+     */
+    public enum Case {
+        LOWER,
+        UPPER,
+        NONE;
+
+        public String wrap(String toWrap) {
+            if (this.equals(NONE)) {
+                return toWrap;
+            }
+
+            return String.format("%s(%s)", this.name(), toWrap);
+        }
+
+        public String wrap(FilterPredicate.FilterParameter toWrap) {
+            return wrap(toWrap.getPlaceholder());
+        }
+    }
+
+    /**
+     * The number of values that are passed to a filter predicate.
+     */
+    public enum ArgumentCount {
+        ZERO,
+        ONE,
+        MANY
+    }
+
+    /**
+     * Constructor.
+     * @param jpqlTemplate A JPQL Query Fragment Template
+     * @param upperOrLower UPPER, LOWER, or NONE
+     * @param argumentCount ZERO, ONE, or MANY
+     */
+    public CaseAwareJPQLGenerator(String jpqlTemplate, Case upperOrLower, ArgumentCount argumentCount) {
+        this.upperOrLower = upperOrLower;
+        this.jpqlTemplate = jpqlTemplate;
+        this.argumentCount = argumentCount;
+    }
+
+    @Override
+    public String generate(String columnAlias, List<FilterPredicate.FilterParameter> params) {
```

Comment:
I think we need a test for `CaseAwareJPQLGenerator` and this method in particular

Reply:
I didn't add an explicit test because this logic is already indirectly covered in all the filter operator tests.  I didn't really write new code - just refactored it.